sympy/solvers/ode.py

```diff
@@ -2964,13 +2984,16 @@ def _constant_renumber(expr):
             sortedargs.sort(key=sort_key)
             return expr.func(*[_constant_renumber(x) for x in sortedargs])
     expr = _constant_renumber(expr)
+
+    # Don't renumber symbols present in the ODE.
+    constants_found = [c for c in constants_found if c not in variables]
```

Comment:
You can use `list(filter_symbols(constants_found, variables))` -- avail from iterables.

Reply:
I don't think that's an improvement. I can see why `filter_symbols` exists (for the infinite iterator situation) but for the finite case a list comprehension is clearer.

Reply:
Okay I guess the list comprehension would be quadratic if variables was a list rather than a set. It always a set as called currently but someone might call it with a list (as shown in the docstring) so I should either use `filter_symbols` or ensure that it is a set.

Reply:
Since I was already calling `set(variables)` above I just changed it to `variables = set(variables)` so that it will always be a set.